common/core/web/input-processor/src/text/prediction/languageProcessor.ts

```diff
@@ -137,17 +137,29 @@ namespace com.keyman.text.prediction {
       let source = specType == 'file' ? model.path : model.code;
       let lp = this;
 
-      // We should wait until the model is successfully loaded before setting our state values.
+      // We pre-emptively emit so that the banner's DOM elements may update synchronously.
+      // Prevents an ugly "flash of unstyled content" layout issue during keyboard load
+      // on our mobile platforms when embedded.
+      lp.currentModel = model;
+      lp.emit('statechange', 'active');
+
       return this.lmEngine.loadModel(source, specType).then(function(config: Configuration) { 
-        lp.currentModel = model;
         lp.configuration = config;
-
-        try {
-          lp.emit('statechange', 'active');
-        } catch (err) {
-          // Does this provide enough logging information?
-          console.error("Could not load model '" + model.id + "': " + (err as Error).message);
+        lp.emit('statechange', 'configured');
+      }).catch(function(error) { 
```

Comment:
The big change + fix.

In particular, note `lp.emit('statechange', 'active');`, which emits an event that triggers display of the OSK's suggestion banner.

Originally, this event was only emitted once a model fully initialized... which requires deferment through a `Promise`.  That deferment is the cause of the 'layout splashing' effect that's been seen in our Android & iOS keyboards for a while now, as the banner simply _will not_ be available for display at the same time as the main body of the OSK.

By instead pre-emptively declaring a model `active` (which needs the associated `currentModel` assignment), we can synchronously set the banner.  Doing so naively does fail to produce initial suggestions, which is what the new `'configured'` state-change event flag is for.